service/dap/server.go

```diff
@@ -754,6 +760,44 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
+	if mode == "replay" {
+		traceDirPath, _ := request.Arguments["traceDirPath"].(string)
```

Comment:
Is program arg required?

Reply:
No, for this path only mode and traceDirPath altough I see your point, the current validation at line 755 will disable core/replay launches even for cases without program but with required args

I don´t imagine scenarios in which program is empty (as it is always autogenerated by vs code) but this is definitely a bug. We may add an exclusion for core and replay modes on the program arg validation

Reply:
Please do have a condition for program arg that excludes the modes that it is not needed for. Thank you.